aips/create_dip.py

```diff
@@ -0,0 +1,326 @@
+#!/usr/bin/env python
+"""
+Create DIP from AIP
+
+Downloads an AIP from the Storage Service and creates a DIP
+"""
+
+import argparse
+import logging
+import logging.config  # Has to be imported separately
+import os
+import sys
+import subprocess
+import shutil
+import uuid
+
+import metsrw
+
+from transfers import amclient
+
+THIS_DIR = os.path.abspath(os.path.dirname(__file__))
+LOGGER = logging.getLogger('create_dip')
+
+
+def setup_logger(log_file, log_level='INFO'):
+    """Configures the logger to output to console and log file"""
+    if not log_file:
+        log_file = os.path.join(THIS_DIR, 'create_dip.log')
+
+    CONFIG = {
+        'version': 1,
+        'disable_existing_loggers': True,
+        'formatters': {
+            'default': {
+                'format': '%(levelname)-8s  %(asctime)s  %(message)s',
+                'datefmt': '%Y-%m-%d %H:%M:%S',
+            },
+        },
+        'handlers': {
+            'console': {
+                'class': 'logging.StreamHandler',
+                'formatter': 'default',
+            },
+            'file': {
+                'class': 'logging.handlers.RotatingFileHandler',
+                'formatter': 'default',
+                'filename': log_file,
+                'backupCount': 2,
+                'maxBytes': 10 * 1024,
+            },
+        },
+        'loggers': {
+            'create_dip': {
+                'level': log_level,
+                'handlers': ['console', 'file'],
+            },
+        },
+    }
+
+    logging.config.dictConfig(CONFIG)
+
+
+def main(ss_url, ss_user, ss_api_key, aip_uuid, tmp_dir, output_dir):
+    LOGGER.info('Starting DIP creation from AIP: %s', aip_uuid)
+
+    if not os.path.isdir(tmp_dir):
+        LOGGER.error('%s is not a valid temporary directory', tmp_dir)
+        return
+
+    if not os.path.isdir(output_dir):
+        LOGGER.error('%s is not a valid output directory', output_dir)
+        return
+
+    # Create empty workspace directory
+    tmp_dir = os.path.join(tmp_dir, aip_uuid)
+    if os.path.exists(tmp_dir):
```

Comment:
Should this be done only if `--overwrite` is passed? `shutil.rmtree` can be highly destructive so I would say any protection is good in production environments.

Reply:
This is a directory created in the temporary directory with the AIP UUID, to download and extract the AIP and we need it empty to avoid mismatches looking for the AIP folder after the extraction. In fact, I was thinking on deleting it at the end of the script and forgot to do it.

Reply:
Ok!